alpaca/data/models/base.py

```diff
@@ -20,12 +21,17 @@ def df(self) -> DataFrame:
         data_list = list(itertools.chain.from_iterable(self.dict().values()))
 
         df = pd.DataFrame(data_list)
+        columns = df.columns
 
         # set multi-level index
         if "news" in self.dict():
             # level=0 - id
             df = df.set_index(["id"])
-        if set(["symbol", "timestamp"]).issubset(df.columns):
+        elif "corporate_action_type" in columns:
+            # level=0 - corporate_action_type
+            df = df.set_index(["corporate_action_type"])
+            df = df.replace({np.nan: None})
```

Comment:
felt np.nan is natural in pandas/numpy context rather than None. Especially for float data type.

Reply:
My problem with NA is that it's often not a number. E.g. a date or a boolean. None feels more natural to me.

Reply:
But I'm happy to remove this line if you feel otherwise.

Reply:
Right. Have got below results. but the auto typing would be depending on version and data.
```python
>>> import pandas as pd
>>> pd.__version__
'2.0.3'
>>> pd.DataFrame([{"bool": True, "date": pd.Timestamp("2023-01-03"), "date_str": "2023-01-03", "str": "", "number": 32, "number_str": "33"}, {"bool": None, "date": None, "date_str": None, "str": None, "number": None, "number_str": None}])
   bool       date    date_str   str  number number_str
0  True 2023-01-03  2023-01-03          32.0         33
1  None        NaT        None  None     NaN       None
```

I usually convert type explicitly to handle time-series data. But wondering we will hire this way or not inside of alpaca-py.
```
>>> df["date_str"] = pd.to_datetime(df["date_str"], format="%Y-%m-%d")
>>> df["number_str"] = df["number_str"].astype(np.float32)
>>> df
   bool       date   date_str   str  number  number_str
0  True 2023-01-03 2023-01-03          32.0        33.0
1  None        NaT        NaT  None     NaN         NaN
```